wpr1:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 30  
  
  # Position Controllers ---------------------------------------
  base_to_torso_position_controller:
    type: position_controllers/JointPositionController
    joint: base_to_torso
    pid: {p: 1000.0, i: 150.0, d: 50.0, i_clamp_max: 500.0, i_clamp_min: -500.0}

  torso_to_upperarm_position_controller:
    type: position_controllers/JointPositionController
    joint: torso_to_upperarm
    pid: {p: 100.0, i: 0.01, d: 10.0}

  upperarm_to_forearm_position_controller:
    type: position_controllers/JointPositionController
    joint: upperarm_to_forearm
    pid: {p: 100.0, i: 0.01, d: 10.0}

  forearm_to_palm_position_controller:
    type: position_controllers/JointPositionController
    joint: forearm_to_palm
    pid: {p: 100.0, i: 0.01, d: 10.0}

  palm_left_finger_position_controller:
    type: position_controllers/JointPositionController
    joint: palm_left_finger
    pid: {p: 100.0, i: 0.01, d: 10.0}
  
  left_finger_tip_position_controller:
    type: position_controllers/JointPositionController
    joint: left_finger_tip
    pid: {p: 10.0, i: 0.01, d: 1.0}

  palm_right_finger_position_controller:
    type: position_controllers/JointPositionController
    joint: palm_right_finger
    pid: {p: 100.0, i: 0.01, d: 10.0}

  right_finger_tip_position_controller:
    type: position_controllers/JointPositionController
    joint: right_finger_tip
    pid: {p: 10.0, i: 0.01, d: 1.0}

  
  #PID Gains ---------------------------------------
  gazebo_ros_control:
     pid_gains:
      base_to_torso:
          p: 1000.0
          i: 150.0
          d: 50.0
          i_clamp_max: 500.0
          i_clamp_min: -500.0
      torso_to_upperarm:
          p: 100.0
          i: 0.01
          d: 10.0
      upperarm_to_forearm:
          p: 100.0
          i: 0.01
          d: 10.0
      forearm_to_palm:
          p: 100.0
          i: 0.01
          d: 10.0
      palm_left_finger:
          p: 100.0
          i: 0.01
          d: 7.0
      left_finger_tip:
          p: 10.0
          i: 0.01
          d: 0.5
      palm_right_finger:
          p: 100.0
          i: 0.01
          d: 7.0
      right_finger_tip:
          p: 10.0
          i: 0.01
          d: 0.5